File: chroma-league/src/main/java/com/bonepl/chromaleague/hud/colors/HextechColor.java
package com.bonepl.chromaleague.hud.colors;

import com.bonepl.razersdk.color.Color;
import com.bonepl.razersdk.color.MultiTransitionColor;
import com.bonepl.razersdk.color.StaticColor;

import java.security.SecureRandom;

public class HextechColor implements Color {
    final MultiTransitionColor hextechColor = new MultiTransitionColor.Builder(StaticColor.BLUE)
            .addTransition(StaticColor.BLUE, 20)
            .addTransition(StaticColor.WHITE, 3)
            .looped(3)
            .build();

    public HextechColor() {
        int random = new SecureRandom().nextInt(hextechColor.getTotalTransitions());
        for (int i = 0; i < random; i++) {
            hextechColor.getColor();
        }
    }

    @Override
    public StaticColor getColor() {
        return hextechColor.getColor();
    }
}

File: src/hud/colors/HextechColor.ts
import {Color} from '../../razer-sdk/color/Color.js';
import {MultiTransitionColor} from '../../razer-sdk/color/MultiTransitionColor.js';
import {StaticColor} from '../../razer-sdk/color/StaticColor.js';

export class HextechColor implements Color {
    readonly hextechColor = new MultiTransitionColor.Builder(StaticColor.BLUE)
        .addTransition(StaticColor.BLUE, 20)
        .addTransition(StaticColor.WHITE, 3)
        .looped(3)
        .build();

    constructor() {
        const random = Math.trunc(Math.random() * this.hextechColor.getTotalTransitions());
        for (let i = 0; i < random; i++) {
            this.hextechColor.getColor();
        }
    }

    getColor(): StaticColor {
        return this.hextechColor.getColor();
    }
}
